src/ApiCore/AuthWrapper.php

```diff
@@ -57,15 +60,49 @@ public function __construct(FetchAuthTokenInterface $fetchAuthTokenInterface, ca
      */
     public function getBearerString()
     {
-        return 'Bearer ' . $this->getToken();
+        return 'Bearer ' . self::getToken($this->fetchAuthTokenInterface, $this->authHttpHandler);
+    }
+
+    /**
+     * @return callable Callable function that returns an authorization header.
+     */
+    public function getAuthorizationHeaderCallback()
+    {
+        $fetchAuthTokenInterface = $this->fetchAuthTokenInterface;
+        $authHttpHandler = $this->authHttpHandler;
+
+        // NOTE: changes to this function should be treated carefully and tested thoroughly. It will
+        // be passed into the gRPC c extension, and changes have the potential to trigger very
+        // difficult-to-diagnose segmentation faults.
+        return function () use ($fetchAuthTokenInterface, $authHttpHandler) {
+            return ['authorization' => ['Bearer ' . self::getToken($fetchAuthTokenInterface, $authHttpHandler)]];
```

Comment:
You should be able to access instance variables in an anonymous function, no need for the use statement (or was this part of the reason we were seeing odd behavior? :P).

Reply:
http://php.net/manual/en/functions.anonymous.php#example-167

I just tested up to PHP 7.2 and it works.

> Closures may also inherit variables from the parent scope. Any such variables must be passed to the use language construct.

The parent scope is the scope in which the anonymous function is defined, which I believe is considered different from the instance scope.

If you'd like to keep it as is because the IDE is complaining, SGTM.

Reply:
Ok, I see what you're saying. I thought you were saying that we can access the local variables without use, not using `$this`. I do want to avoid accessing the AuthWrapper object from the function, even as `$this`, because that seemed to be the cause of the problem.